src/components/Redirect.jsx

```diff
@@ -0,0 +1,109 @@
+import React from 'react'
+
+import { withClient } from 'cozy-client'
+import Intents from 'cozy-interapp'
+import { Button } from 'cozy-ui/react/Button'
+import Input from 'cozy-ui/react/Input'
+import Textarea from 'cozy-ui/react/Textarea'
+import Label from 'cozy-ui/react/Label'
+
+class Redirect extends React.Component {
+  constructor() {
+    super()
+    this.state = {
+      doctype: localStorage.getItem('customDoctype'),
+      options: this.parseAndHandleError(localStorage.getItem('customOptions')),
+      parsingError: false
+    }
+
+    this.handleChange = this.handleChange.bind(this)
+    this.handleSubmit = this.handleSubmit.bind(this)
+  }
+
+  handleChange(ev) {
+    const value = ev.target.value
+    switch (ev.target.name) {
+      case 'doctype':
+        this.setState({ doctype: value })
+        localStorage.setItem('customDoctype', value)
+        break
+      case 'options': {
+        const parsed = this.parseAndHandleError(value)
+        this.setState({ options: parsed })
+        localStorage.setItem('customOptions', JSON.stringify(parsed))
+        break
+      }
+      default:
+        return
+    }
+  }
+
+  handleSubmit() {
+    this.redirect()
+  }
+
+  parseAndHandleError(toParse) {
+    if (!toParse) return null
+    try {
+      const parsed = JSON.parse(toParse)
+      this.setState({
+        parsingError: false
+      })
+      return parsed
+    } catch (e) {
+      this.setState({
+        parsingError: true
+      })
+      return toParse
+    }
```

Comment:
This method is called in the constructor. I am not sure that doing setState in the constructor is supported.